src/rt/dmain2.d

```diff
@@ -472,8 +482,27 @@ extern (C) int _d_run_main(int argc, char **argv, MainFunc mainFunc)
     //       thrown during cleanup, however, will abort the cleanup process.
     void runAll()
     {
-        if (rt_init() && runModuleUnitTests())
-            tryExec({ result = mainFunc(args); });
+        if (rt_init())
+        {
+            auto utResult = runModuleUnitTests();
+            assert(utResult.passed <= utResult.executed);
+            if(utResult.passed == utResult.executed)
+            {
+
+                if (utResult.summarize)
+                    .fprintf(.stderr, "%d unittests passed\n", cast(int)utResult.passed);
+                if(utResult.runMain)
+                    tryExec({ result = mainFunc(args); });
+                else
+                    result = EXIT_SUCCESS;
+            }
+            else
+            {
+                if (utResult.summarize)
+                    .fprintf(.stderr, "%d/%d unittests FAILED\n", cast(int)(utResult.executed - utResult.passed), cast(int)utResult.executed);
```

Comment:
Most test frameworks will print the same summary regardless if something failed or not. That is, they usually print the total number of tests, the number of passed and the number of failed. I'm not sure how we would like it to behave.

Reply:
If nothing failed, then the number of tests run and passed is printed (a few lines above this).

This behavior is totally up for grabs if someone wants to make the effort to repaint the bikeshed. My goal is simply to stop the runtime from always executing main after running unit tests.